scripts/release-binary.sh

```diff
@@ -64,8 +64,10 @@ fi
 
 # Make sure the release binaries are built on x86_64 Ubuntu 16.04 (Xenial)
 if [ "${CHECK}" -eq 1 ]; then
-  UBUNTU_RELEASE=${UBUNTU_RELEASE:-$(lsb_release -c -s)}
-  [[ "${UBUNTU_RELEASE}" == 'xenial' ]] || { echo 'Must run on Ubuntu 16.04 (Xenial).'; exit 1; }
+  if [[ "${BAZEL_BUILD_ARGS}" != *"--config=remote-"* ]]; then
```

Comment:
Sorry, but this is wrong. Why did we need this change? Per discussion in istio/proxy#2569, we need to keep building binaries on Ubuntu 16.04, and RBE shouldn't be excluded from that, since those are the binaries that we publish.

Reply:
It is prow that runs this script right? I think if rbe is configured, local builder image os version should not matter.

Reply:
@PiotrSikora binary linked at RBE is targeted at what glibc image for the RBE image (in this case Xenial) so it doesn't matter what local image is. #2574  is the real fix for RBE postsubmit though.

Reply:
Do we have a way to guard and verify this for RBE images? I really don't want binary requirements to be accidentally "upgraded", as it happens from time to time.

Reply:
@PiotrSikora I agree that we should do that. However the point of avoiding this check is that if we're doing RBE build, the binary requirement has nothing to do with the local builder image.

Reply:
@lizan agreed, my point is that instead of checking RBE and not local builder image, we completely disabled this check in RBE builds.